.github/workflows/check-pull-request.yml

```diff
@@ -40,47 +37,106 @@ jobs:
       - run: cargo +nightly fmt --all -- --check
 
   lint:
-    if: ${{ !startsWith(github.head_ref, 'release/') }}
     name: Clippy
     runs-on: ubuntu-latest
     continue-on-error: false
     steps:
       - uses: actions/checkout@v4
       - run: ./.github/scripts/free_disk_space.sh
       - run: sudo apt-get install -y protobuf-compiler
-      - uses: actions/cache@v4
+
+      - name: Fetch cache
+        uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43 # v2.7.0
+        with:
+          # consistent cache across jobs
+          shared-key: "bajun-cache-cargo-debug"
+
+      - run: cargo clippy --workspace --all-features --all-targets -- -D warnings
+        env:
+          SKIP_WASM_BUILD: 1
+
+  cargo-zepter:
+    name: Cargo Zepter
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install stable Rust
+        uses: actions-rs/toolchain@v1
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-lint-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
I think this corresponds to the default that `rust-cache` does internally, why did you add this?

Reply:
Honestly, It was added before I even entered Ajuna, never paid much attention to it